lib/engine-universal/src/code_memory.rs

```diff
@@ -109,12 +109,12 @@ impl CodeMemory {
             executable_section_result.push(s);
         }
 
-        self.start_of_nonexecutable_pages = bytes;
+        self.start_of_nonexecutable_pages = Mmap::round_up_to_page_size(bytes);
```

Comment:
Should we zero-out the executable leftover?

Reply:
I don't think it is necessary to zero out this mmap at all. Due to wasm using a Harvard-style execution model, the leftover memory isn't going to be accessible to the contract. Unless we mess up our VM implementation, in which we probably have worse problems (e.g. contracts getting access to a source of non-determinism or private counterparts of keys stored in memory)

Reply:
Agree that this isn't *necessary*, but might still be good for defense in depth -- it just feels scary to have an executable memory somewhere in process, whose contents is not explicitly set. Though, this is very weak opinion. If we do enable huge tables, we potentially could need to zero-out a significant chunk of stuff...